### PR TITLE
Integrate SQ in Jenkins

### PR DESCRIPTION
**Pull request by amelie-renard-sonarsource**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            steps {
+                checkout scm
+            }
+        }
+
+        stage('Download Build Wrapper') {
+            steps {
+                powershell '''
+                  $path = "$HOME/.sonar/build-wrapper-win-x86.zip"
+                  rm build-wrapper-win-x86 -Recurse -Force -ErrorAction SilentlyContinue
```
Comment (tomasz-kaminski-sonarsource) [inline Jenkinsfile]: We are removing the dirs, as if they could exits from previous builds, but in the cmake we just call build. Is there a some special meaning for $HOME (so it is not cleaned between jobs), and different from the normal working directory of jenkins?
If so, maybe create .sonar dir not in home, but working directory?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                  Add-Type -AssemblyName System.IO.Compression.FileSystem
+                  [System.IO.Compression.ZipFile]::ExtractToDirectory($path, "$HOME/.sonar")
+                '''
+            }
+        }
+
+        stage('Build') {
+            steps {
+                powershell '''
+                  $env:Path += ";$HOME/.sonar/build-wrapper-win-x86"
```
Comment (tomasz-kaminski-sonarsource) [inline Jenkinsfile]: Maybe move this update of Path afete downloading build-wrapper? That I would believe will make things more localized.

File: Jenkinsfile (hunk 1 above)
Comment (amelie-renard-sonarsource) [reply Jenkinsfile]: You're right, that's a mistake. I'll correct it

File: Jenkinsfile [hunk 3]
```diff
@@ -11,9 +11,9 @@ pipeline {
             steps {
                 powershell '''
                   $path = "$HOME/.sonar/build-wrapper-win-x86.zip"
-                  rm build-wrapper-win-x86 -Recurse -Force -ErrorAction SilentlyContinue
+                  rm $HOME/.sonar/build-wrapper-win-x86 -Recurse -Force -ErrorAction SilentlyContinue
```
Comment (tomasz-kaminski-sonarsource) [inline Jenkinsfile]: Why we still needs to have this rms?

File: Jenkinsfile [hunk 4]
```diff
@@ -11,9 +11,9 @@ pipeline {
...
                   rm $path -Force -ErrorAction SilentlyContinue
-                  mkdir $HOME/.sonar
+                  New-Item -ItemType directory -Path .sonar -Force
                   [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
                   (New-Object System.Net.WebClient).DownloadFile("<SonarQube URL>/static/cpp/build-wrapper-win-x86.zip", $path) <# Replace with your SonarQube server URL #>
                   Add-Type -AssemblyName System.IO.Compression.FileSystem
```
Comment (tomasz-kaminski-sonarsource) [inline Jenkinsfile:23]: I think you are still unpacking the files into home dir, to local .sonar.

**Review by tomasz-kaminski-sonarsource — APPROVED** [review]

**Pull request merged**